west/main.py

```diff
@@ -61,6 +61,8 @@ def parse_args(argv):
         parser = command.add_parser(subparser_gen)
         parser.set_defaults(handler=partial(command_handler, command))
 
+    if argv is None:
+        argv = sys.argv[1:]
     args, unknown = west_parser.parse_known_args(args=argv)
```

Comment:
Do you need to pass the argument list to `parse_known_args()`? Without one, it'll implicitly use `sys.argv` (and will get the command name from `sys.argv[0]`).

Never used subparsers, so might have missed something.

Reply:
If it's to be able to pass a different argument list for testing, maybe this would work:

```python
def parse_args(argv=sys.argv):
    ...
    args, unknown = west_parser.parse_known_args(argv)
```

Reply:
As a style thing, I prefer not to use mutable values as kwarg defaults. These are evaluated only once, rather than at each invocation of the routine, so any inadvertent modifications can subtly change behavior.

Reply:
Alright, fair enough.